new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainInPV.java

```diff
@@ -573,20 +596,291 @@ public void testDomainInPvUsingWdt() {
+                    .mountPath("/shared")
+                    .name(pvName)))
+            .adminServer(new AdminServer() //admin server
+                .serverStartState("RUNNING")
+                .adminService(new AdminService()
+                    .addChannelsItem(new Channel()
+                        .channelName("default")
+                        .nodePort(0))
+                    .addChannelsItem(new Channel()
+                        .channelName("T3Channel")
+                        .nodePort(t3ChannelPort))))
+            .addClustersItem(new Cluster() //cluster
+                .clusterName(clusterName)
+                .replicas(replicaCount)
+                .serverStartState("RUNNING")));
+
+    // verify the domain custom resource is created
+    createDomainAndVerify(domain, introDomainNamespace);
+
+    // verify admin server pod is ready
+    checkPodReady(adminServerPodName, domainUid, introDomainNamespace);
+
+    // verify the admin server service created
+    checkServiceExists(adminServerPodName, introDomainNamespace);
+
+    // verify managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Waiting for managed server pod {0} to be ready in namespace {1}",
+          managedServerPodNamePrefix + i, introDomainNamespace);
+      checkPodReady(managedServerPodNamePrefix + i, domainUid, introDomainNamespace);
+    }
+
+    // verify managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Checking managed server service {0} is created in namespace {1}",
+          managedServerPodNamePrefix + i, introDomainNamespace);
+      checkServiceExists(managedServerPodNamePrefix + i, introDomainNamespace);
     }
+
+    logger.info("change the cluster size and verify the introspector runs and updates the domain status");
+    // create a temporary WebLogic WLST property file
+    File wlstPropertiesFile = assertDoesNotThrow(() -> File.createTempFile("wlst", "properties"),
+        "Creating WLST properties file failed");
+    Properties p1 = new Properties();
+    p1.setProperty("admin_host", K8S_NODEPORT_HOST);
+    p1.setProperty("admin_port", Integer.toString(t3ChannelPort));
+    p1.setProperty("admin_username", ADMIN_USERNAME_DEFAULT);
+    p1.setProperty("admin_password", ADMIN_PASSWORD_DEFAULT);
+    p1.setProperty("cluster_name", clusterName);
+    p1.setProperty("test_name", "change_server_count");
+    assertDoesNotThrow(() -> p1.store(new FileOutputStream(wlstPropertiesFile), "wlst properties file"),
+        "Failed to write the WLST properties to file");
+
+    // change the server count of the cluster by running online WLST
+    Path configScript = Paths.get(RESOURCE_DIR, "python-scripts", "introspect_version_script.py");
+    executeWLSTScript(configScript, wlstPropertiesFile.toPath(), introDomainNamespace);
+
+    // construct a patch string to update introspectVersion
+    StringBuffer patchStr = new StringBuffer("[{")
```

Comment:
Here we can use the logic used in TestActions.patchDomainResourceWithNewRestartVersion to dynamically generate the introspectVersion string, instead of hard-coding.

This utility can be moved to TestActions class as patchDomainResourceWithNewIntrospectVersion

Reply:
Fixed